### PR TITLE
Document traceId/spanId injection into logs

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**
Resolves https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/1249

### CODE REVIEW

File: docs/logger-mdc-instrumentation.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+The OTel Java agent injects several pieces of information about the current span into each logging
+event's MDC copy:
+
+- `traceId` - the current trace id
+  (same as `TracingContextUtils.getCurrentSpan().getContext().getTraceIdAsHexString()`);
+- `spanId` - the current span id
+  (same as `TracingContextUtils.getCurrentSpan().getContext().getSpanIdAsHexString()`);
+- `sampled` - a boolean flag marking whether the current span is sampled or not
+  (same as `TracingContextUtils.getCurrentSpan().getContext().isSampled()`).
+  Only added when `sampled` is `true`.
```
Comment (iNikem) [inline docs/logger-mdc-instrumentation.md:20]: If it is always true, why include it?

Comment (mateuszrzeszutek) [reply docs/logger-mdc-instrumentation.md]: I've added an explanation to the doc. Basically if it's there the span is sampled, if not it is not - I guess the reasoning behind it is that most of spans are not sampled and keeping sampled=false in every log line would be just a waste of space.

**Review by iNikem — APPROVED** [review]

File: docs/logger-mdc-instrumentation.md (hunk 1 above)
Comment (iNikem) [reply docs/logger-mdc-instrumentation.md]: Aren't most span sampled? With `AlwaysOn` sampler, which is default

Comment (mateuszrzeszutek) [reply docs/logger-mdc-instrumentation.md]: 🤦  That's true...

I somehow thought that it worked completely the other way 😅 

Now I'm starting to wonder why it works that way... @anuraaga I think you've added it first in the logback instrumentation -- why do we only log `true` value?

**Review by trask — APPROVED** [review]
Nice doc 👍

Good question about whether we want to change `sampled` field behavior, it seems like it would be easier for log parsers if the field is not conditional(?)

**Comment by anuraaga** [comment]
Yeah it's just a premature optimization, I don't think I taught too deeply when making that change. Always adding sampled seems nice to me too.

**Review by anuraaga — APPROVED** [review]

**Comment by iNikem** [comment]
> Yeah it's just a premature optimization, I don't think I taught too deeply when making that change. Always adding sampled seems nice to me too.

Does this mean we have to make some code change in our MDC handling?

**Comment by anuraaga** [comment]
@iNikem Yeah - so I'm good with merging this first and updating code / README in a followup.

**Comment by mateuszrzeszutek** [comment]
Hmm, looks like random errors:

> Could not GET 'https://jcenter.bintray.com/org/springframework/spring-beans/5.0.9.RELEASE/spring-beans-5.0.9.RELEASE.pom'. Received status code 502 from server: Bad Gateway
> Could not GET 'https://jcenter.bintray.com/io/projectreactor/ipc/reactor-netty/0.7.8.RELEASE/reactor-netty-0.7.8.RELEASE.pom'. Received status code 502 from server: Bad Gateway
> Could not GET 'https://jcenter.bintray.com/org/springframework/spring-beans/5.2.1.RELEASE/spring-beans-5.2.1.RELEASE.pom'. Received status code 502 from server: Bad Gateway

**Comment by mateuszrzeszutek** [comment]
Rebased to force CircleCI run

**Comment by iNikem** [comment]
@mateuszrzeszutek will you do a follow-up fix right away or will you create an issue for that?

**Comment by mateuszrzeszutek** [comment]
I'll do it right away.

**Pull request merged**